sound/soc/sof/ipc4-pcm.c

```diff
@@ -201,6 +202,9 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	snd_mask_none(fmt);
 	snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S32_LE);
 
+	rate->min = ipc4_copier->data.out_format.sampling_frequency;
+	rate->max = rate->min;
+
```

Comment:
I have zero understanding of this patch. Which copier are we talking about here and where is the SRC inserted in the topology? We currently don't have one

Reply:
@RanderWang Are you saying that in the non-SRC case, the hw_params are propagated from FE to BE, so everything works as long as the FE and BE use the same rate, but if the FE hw_params are different to the BE params, we don't propagate the hw_params any longer?

Reply:
yes, exactly

Reply:
First I wanted to use "ipc4_copier->data.out_format.sampling_frequency;" but later I found that it is zero at this time. The reason is that dai_fixup happens before  pcm_hw_params (please check ASoC) in which copier->out_format will be prepared. So I have to two choice (1) hard code to 48K which is actually used by IPC3 (2) use available sampling rate